### PR TITLE
Fixed #30673 -- Relaxed system check for db_table collision when database routers are installed by turning the error into a warning.

### PR DESCRIPTION
**Pull request by uadnan**
https://code.djangoproject.com/ticket/30673

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot

File: docs/ref/checks.txt [hunk 1]
```diff
@@ -317,6 +317,9 @@ Models
   or a number.
 * **models.E034**: The index name ``<index>`` cannot be longer than
   ``<max_length>`` characters.
+* **models.W035**: ``db_table`` ``<db_table>`` is used by multiple models:
+  ``<model list>``. This may not be an issue because database routers are
+  installed.
```
Comment (carltongibson) [inline docs/ref/checks.txt]: I'm not 100% happy with the wording here. 

Can't think of better at this exact moment. I'll come back to it tomorrow. In the meantime, in anyone has a nice phrasing, that'd be good.

File: django/core/checks/model_checks.py [hunk 2]
```diff
@@ -35,16 +36,30 @@ def check_all_models(app_configs=None, **kwargs):
...
             constraints[model_constraint.name].append(model._meta.label)
-    for db_table, model_labels in db_table_models.items():
-        if len(model_labels) != 1:
-            errors.append(
-                Error(
-                    "db_table '%s' is used by multiple models: %s."
-                    % (db_table, ', '.join(db_table_models[db_table])),
-                    obj=db_table,
-                    id='models.E028',
+    if settings.DATABASE_ROUTERS:
```
Comment (carltongibson) [inline django/core/checks/model_checks.py]: I'd de-duplicate these blocks a bit. (See main comment.)

**Comment by felixxm** [comment]
@carltongibson Thanks, agreed, lets keep the same message and we can add your suggestion as a `hint`, I will update this later today.

**Pull request merged**